### ISSUE TITLE
Datagrid: Modal dialog containing a datagrid with small number of columns the last column doesn't fill the space

### ISSUE THREAD

**Issue by claudenbach** [infor-design/enterprise#3562]
Describe the bug
A modal dialog that contains a datagrid with a limited number of columns, the last column doesn't stretch to fill the space.  This gives the appearance that the modal dialog hasn't completely finished rendering.  

Using the gridOption of stretchColumn is not workable for Landmark.  Landmark will know the actual last column only after data has been retrieved since a column may conditionally be hidden based on the data.

To Reproduce
1. Using the attached example-modal-datagrid-landmark.html
2. Click on the Open Modal button
3. Notice that the last column doesn't extend to fill the space

Expected behavior
Last column fills the space.  

Version
v4.27.0-dev

Additional context
Discussed adding the class 'datagrid-default-modal-width' to the modal-content div, but this is not workable in Angular.  See below for other ideas attempted.

**Comment by claudenbach** [comment]
![image](https://user-images.githubusercontent.com/39804038/75584616-ba161f80-5a35-11ea-9b81-3ebfa34dcd48.png)

**Comment by tmcconechy** [comment]
Somehow the modal needs a width. Is there no way at all one can be provided? Maybe we can give all modal datagrids a default width?

**Comment by claudenbach** [comment]
[Git-3562.zip](https://github.com/infor-design/enterprise/files/4268979/Git-3562.zip)

**Comment by claudenbach** [comment]
Tried adding a cssClass of 'datagrid-default-modal-width' on the sohoModalDialogService but that didn't have an effect.

Trided adding the class of 'datagrid-default-modal-width' onto the div that is the modal placeholder, but this didn't work either.

In angular, I'm not seeing a way to get the 
<div id="modal-content" class="datagrid-default-modal-width">

**Comment by tmcconechy** [comment]
Suppose we could default a width, may not always be correct. Or provide a modal width option. Will take another look

**Comment by claudenbach** [comment]
The SohoModalOptions does have the option of maxWidth?  

I was thinking we could use the option cssClass giving it the value of 'datagrid-default-modal-width' or something else that we agree to.

**Comment by claudenbach** [comment]
Can't test with maxWidth, since its not exposed via SohoModalDialogRef.

**Comment by tmcconechy** [comment]
We can try and expose it but thats more of a stretchWidth for the modal. The actual Datagrid element needs a width. But will look in more details when we can. This is just from what i recall before

**Comment by tmcconechy** [comment]
@claudenbach I have fix for this on the EP side but having a hard time testing it in Angular. Can you help with an NG example that resembles what your doing.

I tried on page http://localhost:4200/ids-enterprise-ng-demo/modal-dialog in the demo app. But here the datagrid is actually visible when rendering the page so the width is calculated correctly and im not seeing what your seeing. So i assume your doing this modal a different way with the datagrid on it?

basically what i did was make the check for `datagrid-default-modal-width` check anywhere in the parent root up so it can be on a wrapper above and work. Couldn't find a better way to default a width without breaking..

```
    <div id="modal-content" class="hidden datagrid-default-modal-width">
      <div id="datagrid">
      </div>
    </div>
```
So any hints on how to reproduce this in Angular? Or at least what way your doing it?

**Comment by janahintal** [comment]
QA Passed. Will now move to Done. Thanks. 
Verified in 
https://master-enterprise.demo.design.infor.com/components/datagrid/test-modal-datagrid-single-column.html?theme=uplift&variant=dark & https://master-enterprise.demo.design.infor.com/components/datagrid/test-modal-datagrid-two-columns.html?theme=soho&variant=dark across all platform. Will restest once beta is deployed.

**Comment by claudenbach** [comment]
@tmcconechy I've attempted to create an example to illustrate the issue we're having in Landmark.  Basically took the existing modal-dialog example and tweaked the modal dialog datagrid component. 

Landmark doesn't set the column width and doesn't specify a stretch column since columns may be hidden once the dataViewSet for the grid is retrieved and loaded.

[DialogWithDatagrid.zip](https://github.com/infor-design/enterprise/files/4482822/DialogWithDatagrid.zip)

**Comment by tmcconechy** [comment]
Thats the fix i made here? Im guessing you were not able to look at it as this is also a more recent fix that the 4.27 release (its in 4.28).

The only requirement now is to add the class "datagrid-default-modal-width" to some container in the parent of the modal. So it gets a size. Nothing else works. The invisible modal needs a size as the grid is rendered while it is closed. So in 4.28 this should work? Did you try it based on that?

https://github.com/infor-design/enterprise/pull/3729/files#diff-e77319aa0dca1c7a736b1fd2a31510c4R5

**Comment by claudenbach** [comment]
I'm still missing something then.  Using the files that are in the zip file.  If I change the file modal-dialog-datagrid.component.html to the following then the datagrid itself is empty?

 
```
<soho-toolbar-flex>
  <soho-toolbar-flex-section>
    <h3>
      <span soho-toolbar-section-title>State List</span>
    </h3>
  </soho-toolbar-flex-section>

  <soho-toolbar-flex-section>
      <button soho-button="icon" icon="open-folder" id="open-btn" class="btn-icon" title="Open" disabled></button>
      <button soho-button="icon" icon="delete" id="delte-btn" class="btn-icon" title="Delete" disabled></button>
      <button soho-button="icon" icon="filter" id="filter-btn" class="btn-icon" title="Filter"></button>
      <button soho-button="icon" icon="refresh" id="refresh-btn" class="btn-icon" title="Refresh"></button>
  </soho-toolbar-flex-section>

  <soho-toolbar-flex more-button class="more toolbar-section">
    <button soho-button="icon" icon="more" id="more-btn" class="btn-actions" title="More"></button>
  </soho-toolbar-flex>
</soho-toolbar-flex>

<div id="caltest" class="hidden datagrid-default-modal-width">
  <div soho-datagrid
       selectable="mixed"
       [treeGrid]="true"
       [columns]="columns"
       [data]="data"
       [filterable]="true"
       [cellNavigation]="true"
       rowHeight="short"
       style="height: 400px">
  </div>
</div>

```

**Comment by claudenbach** [comment]
![image](https://user-images.githubusercontent.com/39804038/79400096-d1a85980-7f4a-11ea-8ec3-195bc57ba665.png)

**Comment by tmcconechy** [comment]
In that code you added just remove hidden:
`<div id="caltest" class="datagrid-default-modal-width">`

Hidden is not needed there as thats deeper inside the modal content thats just for the modal root and is not getting unset there.

There is a wierd thing i saw about this particular example its sizing itself another way. But the hidden is why your getting that result in the previous comment.
